tests/lib/rules/no-controller-access-in-routes.js

```diff
@@ -0,0 +1,209 @@
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-controller-access-in-routes');
+const RuleTester = require('eslint').RuleTester;
+
+const { ERROR_MESSAGE } = rule;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('babel-eslint'),
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+  },
+});
+
+ruleTester.run('no-controller-access-in-routes', rule, {
+  valid: [
+    `
+      import Route from '@ember/routing/route';
+      export default class MyRoute extends Route {
+        setupController(controller, ...args) {
+          super.setupController(controller, ...args);
+          const foo = controller.foo;
+        }
+      }
+    `,
+    `
+      import Route from '@ember/routing/route';
+      export default class MyRoute extends Route {
+        resetController(controller, ...args) {
+          super.resetController(controller, ...args);
+          const foo = controller.foo;
+        }
+      }
+    `,
+    `
+      import Component from '@ember/component';
+      import { action, get } from '@ember/object';
+      export default class MyComponent extends Component {
+        @action
+        myAction() {
+          const controller = this.controller;
+        }
+      }
+    `,
+    {
+      code: `
+        import Route from '@ember/routing/route';
+        import { action } from '@ember/object';
+        export default class MyRoute extends Route {
+          @action
+          myAction() {
+            const controller = this.controllerFor('my');
+          }
+        }
+      `,
+      options: [{ allowControllerFor: true }],
+    },
+  ],
+  invalid: [
+    {
+      code: `
+        import Route from '@ember/routing/route';
+        import { action } from '@ember/object';
+        export default class MyRoute extends Route {
+          @action
+          myAction() {
+            const controller = this.controller;
+          }
+        }
+      `,
+      output: null,
+      errors: [{ message: ERROR_MESSAGE, type: 'MemberExpression' }],
+    },
+    {
+      code: `
+        import Route from '@ember/routing/route';
+        import { action } from '@ember/object';
+        export default class MyRoute extends Route {
+          @action
+          myAction() {
+            const controller = this.controllerFor('my');
```

Comment:
You could add another invalid example that uses `controllerFor` without specifying `options` to ensure `allowControllerFor` is off by default.

Reply:
Added case for `allowControllerFor` default value.